content/patterns/layered-zero-trust/lzt-secure-supply-chain.adoc

```diff
@@ -0,0 +1,557 @@
+
+The process of signing and attaching the SBOM runs directly by using `cosign`, which operates on the image.
+
+.Procedure
+
+* Attach the SBOM to the image:
++
+[source,terminal]
+----
+# Attach the SBOM
+$ cosign attest \
+  --predicate qtodo-sbom.json \
+  --type spdxjson \
+  --fulcio-url="${FULCIO_URL}" \
+  --rekor-url="${REKOR_URL}" \
+  --oidc-client-id="${RHTAS_CLIENT_ID}" \
+  --oidc-issuer "${OIDC_ISSUER_URL}" \
+  --yes \
+  "${IMAGE}"
+----
+
+[id="upload-sbom-rhtpa"]
+== Uploading the SBOM to RHTPA
+
+To analyze the SBOM and detect Common Vulnerabilities and Exposures (CVEs) in the image, use RHTPA.
+
+The RHTPA system provides an API that enables the direct upload of the SBOM by an HTTP request. To use this API, you need a token authorized by the OIDC.
+
+This following commands illustrates the process for obtaining the token when using Keycloak, which serves as the default OIDC for the Zero Trust Validated Pattern (ZTVP).
+
+.Procedure
+
+. Get the OIDC Issuer URL from Keycloak and set the token URL:
++
+[source,terminal]
+----
+# Get the OIDC Issuer URL from Keycloak route
+$ export OIDC_ISSUER_URL="https://$(oc get route -n keycloak-system -l app=keycloak -o jsonpath='{.items[0].spec.host}')/realms/ztvp"
+$ export OIDC_TOKEN_URL="${OIDC_ISSUER_URL}/protocol/openid-connect/token"
+----
+
+. Obtain the client secret for the RHTPA OIDC client:
++
+[source,terminal]
+----
+# Get client secret
+$ export RHTPA_CLIENT_SECRET="$(oc get secret rhtpa-oidc-cli-secret -n trusted-profile-analyzer -o jsonpath='{.data.client-secret}' | base64 -d)"
+----
+
+. Set the client ID:
++
+[source,terminal]
+----
+# Set clientID
+$ export RHTPA_CLIENT_ID="rhtpa-cli"
+----
+
+. Obtain a Keycloak token:
++
+[source,terminal]
+----
+# Obtain a Keycloak token
+$ export RHTPA_TOKEN=$(curl -sSfk -X POST "${OIDC_TOKEN_URL}" \
+  -H 'Accept: application/json' \
+  -H 'Content-Type: application/x-www-form-urlencoded' \
+  -d 'grant_type=client_credentials' \
+  -d "client_id=${RHTPA_CLIENT_ID}" \
+  -d "client_secret=${RHTPA_CLIENT_SECRET}" | jq -r .access_token)
+----
+
+. Get the RHTPA API URL:
++
+[source,terminal]
+----
+# Get the RHTPA API hostname
+$ export RHTPA_URL="https://$(oc get route -n trusted-profile-analyzer -l app.kubernetes.io/name=server -o jsonpath='{.items[0].spec.host}')"
+----
+. Upload the SBOM to RHTPA using the API with the Keycloak token:
++
+[source,terminal]
+----
+# Upload the SBOM using curl
+$ curl -sk -X POST \
+  -H "Authorization: Bearer ${RHTPA_TOKEN}" \
+  -H "Content-Type: application/json" \
+  --data-binary "@qtodo-sbom.json" \
+  "${RHTPA_URL}/api/v2/sbom"
+----
+
+Upon publication, you can view the SBOM on the RHTPA web UI. The access URL is same as the one used for the API. Similar to the API, the RHTPA web UI uses OIDC for user authentication. In our specific deployment, you can view the RHTPA web UI by logging in with the credentials. Use the following commands to get the username and password:
```

Comment:
```suggestion
Once the SBOM has been published, it can be viewed within the RHTPA web UI. The URL is same as the value obtained previously for the RHTPA API. Similar to the API, the RHTPA web UI uses OIDC for user authentication. In our specific deployment, you can view the RHTPA web UI by logging in with the credentials associated with the user. Use the following commands to obtain the username and password:
```